connectivity/FEATURE_BLE/source/cordio/source/PalGapImpl.h

```diff
@@ -364,18 +364,30 @@ class PalGap final : public ble::PalGap {
 
         static GapConnectionCompleteEvent convert(const hciLeConnCmplEvt_t *conn_evt)
         {
+            const bdAddr_t *peer_rpa = &conn_evt->peerRpa;
+            const bdAddr_t *peer_address = &conn_evt->peerAddr;
+
+#if defined(TARGET_MCU_STM32WB55xx)
```

Comment:
Thanks for the feedback @jeromecoutant . The bug is inside the BLE firmware which as far as I know is only proposed as binary. The workaround is *not* in the cordio sources but on top of it, in Mbed OS interfaces with the Cordio stack. The workaround does not affect devices producing correct enhanced connection events. It doesn't require modification if it is fixed in a new firmware by ST. 

I truly believe it is the best place to put the workaround today as users doesn't always update to the latest version of the firmware.Even if ST was releasing a new firmware today, I'd keep the workaround in place to avoid issues with old firmwares.